modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -1196,4 +1198,38 @@ public RequestMappingMode getRequestMappingMode() {
     public void setRequestMappingMode(RequestMappingMode requestMappingMode) {
         this.requestMappingMode = requestMappingMode;
     }
+
+    @Override
+    public CodegenParameter fromParameter( final Parameter parameter, final Set<String> imports ) {
+        CodegenParameter codegenParameter = super.fromParameter( parameter, imports );
+        if( !codegenParameter.isContainer || codegenParameter.isMap){
+            return codegenParameter;
+        }
+        codegenParameter.datatypeWithEnum = replaceBeanValidationCollectionType(codegenParameter.items, codegenParameter.datatypeWithEnum  );
+        codegenParameter.dataType = replaceBeanValidationCollectionType(codegenParameter.items, codegenParameter.dataType  );
+        return codegenParameter;
+    }
+
+    @Override
+    public CodegenProperty fromProperty( String name, Schema p, boolean required, boolean schemaIsFromAdditionalProperties ) {
+        CodegenProperty codegenProperty = super.fromProperty( name, p, required, schemaIsFromAdditionalProperties );
+        if( !codegenProperty.isContainer || codegenProperty.isMap){
+            return codegenProperty;
+        }
+        codegenProperty.datatypeWithEnum = replaceBeanValidationCollectionType(codegenProperty.items, codegenProperty.datatypeWithEnum );
+        codegenProperty.dataType = replaceBeanValidationCollectionType(codegenProperty.items, codegenProperty.dataType  );
+        return codegenProperty;
+    }
+
+    private String replaceBeanValidationCollectionType(CodegenProperty codegenProperty, String dataType) {
+        if (useJakartaEe || !useBeanValidation || !codegenProperty.isModel || codegenProperty.baseName.toLowerCase(Locale.ROOT).contains("response")) {
```

Comment:
This might be worth documenting as well.

Why does Jakarta EE prevent using `@Valid` ?

Reply:
No Jakarta Expert. Have to migrate my project to sping boot 3 .0 :)
but looks like that for jakarta there is no @Valid Annotation anymore.

Reply:
there is `@Valid` annotation under package `jakarta.validation`

https://jakarta.ee/specifications/bean-validation/3.0/apidocs/index.html?jakarta/validation/Valid.html

Reply:
@MelleD I assume that you are trying to fix the issue with `spring http interface`. Just exclude this library with `!SPRING_HTTP_INTERFACE.equals(library)`, because beanValidation isn't applicable for it

Reply:
@borsch Yes this one reason, but do we have a example with Jakarta and @Valid ? Found nothing in the codebase

I read something in hibernate validator that this not longer necessary for container classes, but didn’t try that.

Reply:
> 

Ok i I’ll add this. Thanks :)

Reply:
@MelleD this is a sample with SB3 & jakarta validation - https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/server/petstore/springboot-3/src/main/java/org/openapitools/model/Category.java

Reply:
@borsch thanks. I see there is no @Valid because we have no Bean relation. Should we add this as well?

Reply:
@MelleD there is `@Valid`, but on different model in the same sample project

https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/server/petstore/springboot-3/src/main/java/org/openapitools/model/Pet.java

Reply:
Ah cool sorry didn't see that is the jakarta valid. Perfect then i skip it with the http-interface :)